oidc.go

```diff
@@ -307,3 +313,16 @@ func (j *jsonTime) UnmarshalJSON(b []byte) error {
 	*j = jsonTime(time.Unix(unix, 0))
 	return nil
 }
+
+func unmarshalResp(r *http.Response, body []byte, v interface{}) error {
+	err := json.Unmarshal(body, &v)
+	if err != nil {
+		ct := r.Header.Get("Content-Type")
+		mediaType, _, _ := mime.ParseMediaType(ct)
```

Comment:
I took a look, and the error isn't useful for this. You basically just get errors like `mime: invalid media parameter`, and mediaType is `""` on errors. The error isn't useful to return or add to the error message, and mediaType will still != "application/json" in the error case, thus returning the other error message about failing to parse the media type.